rmw_fastrtps_cpp/src/rmw_publisher.cpp

```diff
@@ -164,6 +173,19 @@ rmw_return_loaned_message_from_publisher(
 rmw_ret_t
 rmw_destroy_publisher(rmw_node_t * node, rmw_publisher_t * publisher)
 {
+  RMW_CHECK_ARGUMENT_FOR_NULL(node, RMW_RET_INVALID_ARGUMENT);
+  RMW_CHECK_ARGUMENT_FOR_NULL(publisher, RMW_RET_INVALID_ARGUMENT);
+  RMW_CHECK_TYPE_IDENTIFIERS_MATCH(
+    node,
+    node->implementation_identifier,
+    eprosima_fastrtps_identifier,
+    return RMW_RET_INCORRECT_RMW_IMPLEMENTATION);
+  RMW_CHECK_TYPE_IDENTIFIERS_MATCH(
+    publisher,
+    publisher->implementation_identifier,
+    eprosima_fastrtps_identifier,
+    return RMW_RET_INCORRECT_RMW_IMPLEMENTATION);
```

Comment:
why doing this checks here.
If they are done in `__rmw_destroy_publisher`, both rmw_fastrtps_cpp and rmw_fastrtps_dynamic_cpp share the same code.

Reply:
That is true, but `rmw_fastrtps_dynamic_cpp` has to do these checks earlier (see [here](https://github.com/ros2/rmw_fastrtps/blob/ece181fa17f3cbcba0bbfccfbd1491b79d972922/rmw_fastrtps_dynamic_cpp/src/rmw_publisher.cpp#L170-L187)). Between having duplicate checks in `rmw_fastrtps_cpp` and `rmw_fastrtps_dynamic_cpp`, or checking the same thing twice in run-time for `rmw_fastrtps_dynamic_cpp`, I chose the former.